src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -211,27 +211,29 @@ private void init() {
   }
 
   private void setCreateGameButtonState() {
-    boolean disable = titleTextField.getText().isEmpty()
-        || featuredModListView.getSelectionModel().getSelectedItem() == null
-        || userService.getConnectionState() != CONNECTED;
+    String title = titleTextField.getText();
 
     ConnectionState lobbyConnectionState = userService.getConnectionState();
-    String createGameButtonText = switch (lobbyConnectionState) {
-      case DISCONNECTED -> i18n.get("game.create.disconnected");
-      case CONNECTING -> i18n.get("game.create.connecting");
+    String createGameTextKey = switch (lobbyConnectionState) {
+      case DISCONNECTED -> "game.create.disconnected";
+      case CONNECTING -> "game.create.connecting";
       case CONNECTED -> {
-        if (Strings.isNullOrEmpty(titleTextField.getText())) {
-          yield i18n.get("game.create.titleMissing");
+        if (Strings.isNullOrEmpty(title)) {
+          yield "game.create.titleMissing";
+        } else if (!StandardCharsets.US_ASCII.newEncoder().canEncode(title)) {
```

Comment:
Need to make a new encoder object everywhere, or can there just be a static final one? Or maybe not static, but just final and init in constructor.

Reply:
Charset encoder is not thread safe

Reply:
Java never ceases to disappoint me :P